decisions/20240313-try-w3up.md

```diff
@@ -0,0 +1,142 @@
+    * initially, this will only happen when a feature switch is enabled
+
+#### Data Model Changes
+
+Minimal. Only new configuration.
+
+We'll need to decide which storage space that uploads will be added to.
+To start, we'll just making everything store in a configured storage space, and the app will read that from configuration passed in environment variables, similar to env vars in [add-to-web3](https://github.com/web3-storage/add-to-web3?tab=readme-ov-file#generating-a-secret_key-and-proof).
+
+##### `W3_NFTSTORAGE_SPACE` Environment Variable
+
+configures the web3.storage space that nfts will be stored in.
+
+Example: `did:key:z6MkfbsERaJ7rtJQWMWtYMxNED56bhQMgrNu8CRdUjB5LfRp`
+
+##### `W3_NFTSTORAGE_PRINCIPAL` Environment Variable
+
+configures how nft.storage will authenticate to web3.storage when sending invocations to store things. e.g. it may contain an ed25519 keypair as a CID with raw multihash.
+
+##### `W3_NFTSTORAGE_PROOF` Environment Variable
+
+configures the capabilities that nft.storage has access to when interacting with web3.storage to store nfts. These capabilities will usually be UCAN delegations whose audience is the identifier of `W3_NFTSTORAGE_PRINCIPAL`.
+W3_NFTSTORAGE_PROOF needs to have proof rooted in W3_NFTSTORAGE_SPACE that authorize W3_NFTSTORAGE_PRINCIPAL to store in W3_NFTSTORAGE_SPACE.
+
+##### `NFTSTORAGE_W3S_ALLOW_ACCOUNTS` Environment Variable
+
+configures feature switch for which nftstorage accounts will have new uploads stored in web3.storage.
+
+Format: JSON Array of Account Identifiers
+
+#### UI Changes
+
+None. But the existing UI workflow of uploading via https://nft.storage/files/ and form should behave just like they do now. But after this change, there should be a new side effect, which is that the upload should appear in the listing of uploads for the configured `W3_NFTSTORAGE_SPACE` (e.g. via w3cli `w3 ls` or in console.web3.storage).
+
+## Risks
+
+### degrading nft.storage's existing functionality
+
+We wouldn't want these changes to have any negative impact on users that rely on nft.storage's current setup. We only want to enable evaluation of new ways of delivering on that functionality that are seamless for end-users.
+
+#### Mitigation: feature switch for initial rollout
+
+We'll deploy this functionality to use web3.storage initially behind an opt-in feature switch. Once deployed, most uploads will not use the web3.storage functionality. Only accounts we add to a configured allow-list will get persisted to w3up. This removes risk of everyday users being impacted by bugs in the new code, while still enabling employees we allowlist to test the code in production. After testing in production, we can add more to the allowlist (e.g. perhaps nft.storage employees), or we can remove the feature switch to roll out the functionality for all users of nft.storage.
```

Comment:
ok, answered my first comment

Reply:
I made a note up above to make it more explicit: https://github.com/nftstorage/nft.storage/pull/2518/commits/8481203c5bbd1da8fc6986b1a5832b0bb615a16b